deployer/update_grafana_datasources.py

```diff
@@ -143,31 +139,54 @@ def build_request_headers():
     return headers
 
 
-def get_clusters_used_as_datasources():
+def get_clusters_used_as_datasources(cluster_name, datasource_endpoint):
     """Returns a list of cluster names that have prometheus instances already defined as datasources of the centralized Grafana."""
-    headers = build_request_headers()
+    headers = build_request_headers(cluster_name)
     # Get a list of all the currently existing datasources
-    response = requests.get(DATASOURCE_ENDPOINT, headers=headers)
+    response = requests.get(datasource_endpoint, headers=headers)
 
     if response.status_code != 200:
         print(
-            f"An error occured when retrieving the datasources from {DATASOURCE_ENDPOINT}. \n Error was {response.text}."
+            f"An error occured when retrieving the datasources from {datasource_endpoint}. \n Error was {response.text}."
         )
         response.raise_for_status()
 
-    print("Successfully retrieved the datasources!")
     datasources = response.json()
-    return [datasource["name"] for datasource in datasources]
+    datasources = [datasource["name"] for datasource in datasources]
+    print_colour(
+        f"Successfully retrieved {len(datasources)} existing datasources! {datasources}"
+    )
+
+    return datasources
 
 
 def main():
+    argparser = argparse.ArgumentParser(
+        description="""A command line tool to update Grafana
+        datasources.
+        """
+    )
+
+    argparser.add_argument(
+        "cluster_name", type=str, help="The name of the cluster where the Grafana lives"
+    )
+
+    argparser.add_argument("grafana_url", type=str, help="The public URL of Grafana")
```

Comment:
Can we read this in from the cluster config too? I think we have it under ingress.

Reply:
Done in https://github.com/2i2c-org/infrastructure/pull/1215/commits/798e53f070b2662acb62404e11a39e140f67d025